integration-tests/main.go

```diff
@@ -0,0 +1,307 @@
+	log.Println("Preparing build context from Dockerfile...")
+	tarball, err := archive.TarWithOptions(config.ProjectDir, &archive.TarOptions{})
+	if err != nil {
+		return err
+	}
+	defer tarball.Close()
+
+	log.Println("Building the Docker image...")
+	resp, err := cli.ImageBuild(ctx, tarball, types.ImageBuildOptions{
+		Tags: []string{config.ImageName},
+	})
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	log.Println("Reading build output...")
+	if _, err := io.Copy(os.Stdout, resp.Body); err != nil {
+		return err
+	}
+
+	log.Println("Docker image built successfully.")
+	return nil
+}
+
+func stopAndRemoveContainer(ctx context.Context, containerName string) error {
+	log.Println("Stopping the existing container...")
+	if err := cli.ContainerStop(ctx, containerName, container.StopOptions{}); err != nil && !client.IsErrNotFound(err) {
+		return err
+	}
+	log.Println("Removing the existing container...")
+	if err := cli.ContainerRemove(ctx, containerName, container.RemoveOptions{}); err != nil && !client.IsErrNotFound(err) {
+		return err
+	}
+	log.Println("Existing container stopped and removed successfully.")
+	return nil
+}
+
+func TearDownTestContainer() error {
+	ctx := context.Background()
+
+	log.Println("Stopping the test container...")
+	if err := cli.ContainerStop(ctx, containerID, container.StopOptions{}); err != nil {
+		return err
+	}
+
+	log.Println("Removing the test container...")
+	if err := cli.ContainerRemove(ctx, containerID, container.RemoveOptions{}); err != nil {
+		return err
+	}
+
+	log.Println("Test container torn down successfully.")
+	return nil
+}
+
+func parseCLIArgs() (ContainerConfig, DbInfoMap, error) {
+	dbConfigStr := flag.String("dbconfig", "", "JSON string of database configuration map")
+	containerConfigStr := flag.String("containerConfig", "", "JSON string of container configuration")
+	flag.Parse()
+
+	var config ContainerConfig
+	if err := json.Unmarshal([]byte(*containerConfigStr), &config); err != nil {
+		return config, DbInfoMap{}, err
+	}
+
+	var dbInfoMap DbInfoMap
+	if err := json.Unmarshal([]byte(*dbConfigStr), &dbInfoMap); err != nil {
+		return config, DbInfoMap{}, err
+	}
+
+	return config, dbInfoMap, nil
+}
+
+// you can run main to spin up a container without running the test suite
+func main() {
+	config, dbInfoMap, err := parseCLIArgs()
+	if err != nil {
+		log.Fatalf("Error parsing CLI args: %v\n", err)
+	}
+
+	dbInfo, ok := dbInfoMap["16.3"] // Change "16.3" to any other version if needed
+	if !ok {
+		log.Fatalf("No DbInfo found for version '16'")
+	}
+
+	log.Printf("Container config: %+v\n", config)
+	log.Printf("DbInfo :  %+v\n", dbInfo)
+	log.Println("RDS Instance: ", dbInfo.AwsRdsInstance)
+	log.Println("DB Connection String: ", dbInfo.DbConnString)
+
+	log.Println("Setting up test container...")
+	if err := SetupTestContainer(&config, dbInfo); err != nil {
+		log.Fatalf("Failed to set up container: %v\n", err)
+	}
+
+	// defer func() {
```

Comment:
Confirming we want to leave this commented out

Reply:
yeah i left this main as a way to spin up the container without running the tests or tearing it down, to help validate things during dev. I guess i could remove the comment

Reply:
Ok. Either way.